cpp/daal/src/data_management/roc_auc_score.cpp

```diff
@@ -0,0 +1,141 @@
+* limitations under the License.
+*******************************************************************************/
+
+#include "data_management/data/internal/roc_auc_score.h"
+#include "data_management/data/numeric_table.h"
+#include "data_management/data/data_dictionary.h"
+#include "services/env_detect.h"
+#include "src/externals/service_dispatch.h"
+#include "src/services/service_data_utils.h"
+#include "src/services/service_utils.h"
+#include "src/services/service_defines.h"
+#include "src/threading/threading.h"
+#include "src/data_management/service_numeric_table.h"
+#include "data_management/features/defines.h"
+#include "src/algorithms/service_error_handling.h"
+#include "src/externals/service_rng.h"
+#include "src/externals/service_rng_mkl.h"
+
+namespace daal
+{
+namespace data_management
+{
+namespace internal
+{
+/*
+template <typename DataType>
+struct Sum 
+{
+    double value;
+
+    Sum() : value(0) {}
+    Sum(Sum& s, tbb::split) : value(0) {}
+
+    void operator()(const tbb::blocked_range<typename std::vector<DataType>::iterator>& r) {
+        value = std::accumulate(r.begin(), r.end(), value);
+    }
+
+    void join(Sum& rhs) { value += rhs.value; }
+};
+*/
+template <typename DataType>
+void calculateRankData(double * predictedRank, NumericTablePtr & prediction_numpy, const int& size)
+{
+    data_management::BlockDescriptor<DataType> numpyBlock;
+    prediction_numpy->getBlockOfRows(0, 1, data_management::readOnly, numpyBlock);
+    DataType * numpyPtr = numpyBlock.getBlockPtr();
+
+    std::pair<DataType, size_t>* v_sort = new std::pair<DataType, size_t>[size];
+
+    for (size_t i = 0; i < size; ++i) {
+        v_sort[i] = std::make_pair(numpyPtr[i], i);
+    }
+
+    for(size_t i = 0;i < size;++i){ //temporarily, bubble sort
+        for(size_t j = i + 1;j < size;++j){
+            if (v_sort[j] < v_sort[i]){
+                std::swap(v_sort[i], v_sort[j]);
+            }
+        }
+    }
+
+    /*
+    tbb::parallel_sort(v_sort.begin(), v_sort.end(), [](auto &left, auto &right) {
+        return left.first < right.first;
+    });
+    */
+
+    int r = 1;
+    int n = 1;
+    size_t i = 0;
+
+    while (i < size) {
+        size_t j = i;
+        while ((j < (size - 1)) && (v_sort[j].first == v_sort[j + 1].first)) {
+            j++;
+        }
+        n = j - i + 1;
+        for (size_t j = 0; j < n; ++j) { // parallel this
+            int idx = v_sort[i+j].second;
+            predictedRank[idx] = r + ((n - 1) * 0.5);
+        }
+        r += n;
+        i += n;
+    }
+    prediction_numpy->releaseBlockOfRows(numpyBlock);
+}
+
+template DAAL_EXPORT void calculateRankData<float> (double * predictedRank, NumericTablePtr & prediction_numpy, const int& size);
+template DAAL_EXPORT void calculateRankData<double>(double * predictedRank, NumericTablePtr & prediction_numpy, const int& size);
+
+template <typename DataType>
+double rocAucScore(double * predictedRank, NumericTablePtr & actual_numpy, const int& size)
```

Comment:
Overall:
1) Need to add CPU dispatcher. Please, refer to https://github.com/oneapi-src/oneDAL/blob/3e3341316628a60f6c5abbf4d36dafc95bd6d344/cpp/daal/src/data_management/finiteness_checker.cpp#L578
2) remove std usage